tests/vptr/basic.cc

```diff
@@ -167,6 +167,99 @@ TEST(pointer_mapper, reuse_ptr) {
     ASSERT_FALSE(PointerMapper::is_nullptr(shouldBeTheSame));
     ASSERT_EQ(pMap.count(), 3u);
     ASSERT_EQ(shouldBeTheSame, reused);
+
+    SYCLfree(shouldBeTheSame, pMap);
+    ASSERT_EQ(pMap.count(), 2u);
+    SYCLfree(end, pMap);
+    ASSERT_EQ(pMap.count(), 1u);
+    SYCLfree(initial, pMap);
+    ASSERT_EQ(pMap.count(), 0u);
```

Comment:
My tests are based on reuse_ptr, for symmetry I also free'd everything here.

Reply:
ok, makes sense